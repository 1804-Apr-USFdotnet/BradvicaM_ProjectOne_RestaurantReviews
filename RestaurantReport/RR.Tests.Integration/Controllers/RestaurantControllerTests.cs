using System.Collections.Generic;
using System.Web.Mvc;
using ApprovalTests;
using ApprovalTests.Reporters;
using Autofac;
using AutoMapper;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using RR.DomainServices;
using RR.Repositories;
using RR.ViewModels;
using RR.Web;
using RR.Web.Controllers;

namespace RR.Tests.Integration.Controllers
{
    [TestClass]
    public class RestaurantControllerTests
    {
        private readonly RestaurantReportTestContext _testContext;
        private readonly RestaurantController _controller;

        public RestaurantControllerTests()
        {
            var container = Bootstrapper.RegisterTypes();
            var mapper = container.Resolve<IMapper>();
            _testContext = new RestaurantReportTestContext();

            _controller = new RestaurantController(new RestaurantService(new RestaurantRepository(_testContext)), mapper);
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void ListRestaurants_OnGet_ReturnsCorrectInformation()
        {
            var result = _controller.ListRestaurants() as ViewResult;

            Approvals.Verify(result.Model.ToString());
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void ListRestaurants_OnPostViewModel_ReturnsCorrectInformation()
        {
            var result = _controller.ListRestaurants(new ListRestaurantsViewModel {ListOrder = "name"}) as ViewResult;

            Approvals.Verify(result.Model.ToString());
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TopRatedRestauarants_OnCall_ReturnsCorrectInformation()
        {
            var result = _controller.TopRatedRestaurants() as ViewResult;

            Approvals.VerifyAll(result.Model as IEnumerable<TopRatedRestaurantViewModel>, "TopRatedRestaurants:");
        }
    }
}
