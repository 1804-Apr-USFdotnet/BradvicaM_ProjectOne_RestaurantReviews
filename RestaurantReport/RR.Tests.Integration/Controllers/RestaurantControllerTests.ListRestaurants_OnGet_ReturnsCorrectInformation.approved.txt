
ListOrder: 
SelectRestaurantPublicId: 00000000-0000-0000-0000-000000000000
SelectListItems: 
name
state
city
rating

ViewRestaurantViewModels: 
RestaurantPublicId: 14534a44-add2-40ef-a173-f5d321b7d2d2
Name: John
Street: 756 You Dr.
City: Claremont
State: CA
ZipCode: 11415
PhoneNumber: 1234567891
AverageRating: 5
Website: www.sfs.com

